cockatrice/src/filterbuilder.cpp

```diff
@@ -8,76 +8,80 @@
 #include "cardfilter.h"
 
 FilterBuilder::FilterBuilder(QWidget *parent)
-    : QWidget(parent)
+	: QWidget(parent)
 {
-    filterCombo = new QComboBox;
-    filterCombo->setObjectName("filterCombo");
-    for (int i = 0; i < CardFilter::AttrEnd; i++)
-        filterCombo->addItem(
-            tr(CardFilter::attrName(static_cast<CardFilter::Attr>(i))),
-            QVariant(i)
-        );
-
-    typeCombo = new QComboBox;
-    typeCombo->setObjectName("typeCombo");
-    for (int i = 0; i < CardFilter::TypeEnd; i++)
-        typeCombo->addItem(
-            tr(CardFilter::typeName(static_cast<CardFilter::Type>(i))),
-            QVariant(i)
-        );
-
-    QPushButton *ok = new QPushButton(QPixmap("theme:icons/increment"), QString());
-    ok->setObjectName("ok");
-    ok->setMaximumSize(20, 20);
-
-    edit = new QLineEdit;
-    edit->setObjectName("edit");
-    edit->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Fixed);
-
-    QGridLayout *layout = new QGridLayout;
-    layout->setObjectName("layout");
-    layout->setContentsMargins(0, 0, 0, 0);
-
-    layout->addWidget(typeCombo, 0, 0, 1, 2);
-    layout->addWidget(filterCombo, 0, 2, 1, 2);
-    layout->addWidget(edit, 1, 0, 1, 3);
-    layout->addWidget(ok, 1, 3);
-
-    setLayout(layout);
-
-    connect(edit, SIGNAL(returnPressed()), this, SLOT(emit_add()));
-    connect(ok, SIGNAL(released()), this, SLOT(emit_add()));
-    fltr = NULL;
+	filterCombo = new QComboBox;
+	filterCombo->setObjectName("filterCombo");
+	for (int i = 0; i < CardFilter::AttrEnd; i++)
+		filterCombo->addItem(
+			tr(CardFilter::attrName(static_cast<CardFilter::Attr>(i))),
+			QVariant(i)
+		);
+
+	typeCombo = new QComboBox;
+	typeCombo->setObjectName("typeCombo");
+	for (int i = 0; i < CardFilter::TypeEnd; i++)
+		typeCombo->addItem(
+			tr(CardFilter::typeName(static_cast<CardFilter::Type>(i))),
+			QVariant(i)
+		);
+
+	QPushButton *ok = new QPushButton(QPixmap("theme:icons/increment"), QString());
+	ok->setObjectName("ok");
+	ok->setMaximumSize(20, 20);
+
+	edit = new QLineEdit;
+	edit->setObjectName("edit");
+	edit->setPlaceholderText(tr("Type your filter here"));
+#if QT_VERSION >= 0x050200
```

Comment:
We build above 5.0.2, so these guards should be removed at this time.

Reply:
I mirrored them from another place in the app.
Will remove both then 👍

Reply:
postponed, see #3005